riscv/src/compiler.rs

```diff
@@ -1184,10 +1182,14 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
                 _ => None,
             };
             match (replacement, instr) {
+                (None, instr) => {
+                    let instr = if instr == "tail" { "jump" } else { instr };
```

Comment:
Can you add a comment saying something like "tail is allowed to use x6 for computation, but does not have to" - is that correct?

Reply:
I am not sure this is true. Strictly speaking, both `tail` and `call` should use it in a very specific way, although no sane program will rely on this behavior.

The thing is, in our current implementation, `call` doesn't do it, and `tail` does it wrong (only the highest bits of the address must be stored in `x6`, not the whole thing).

Reply:
I've added a comment explaining it.